core/src/test/scala/unit/kafka/server/GroupCoordinatorBaseRequestTest.scala

```diff
@@ -267,56 +267,106 @@ class GroupCoordinatorBaseRequestTest(cluster: ClusterInstance) {
     val syncGroupRequest = new SyncGroupRequest.Builder(syncGroupRequestData).build()
     val syncGroupResponse = connectAndReceive[SyncGroupResponse](syncGroupRequest)
     assertEquals(expectedError.code, syncGroupResponse.data.errorCode)
+
+    syncGroupResponse.data
   }
 
-  protected def joinConsumerGroupWithOldProtocol(
+  protected def sendJoinRequest(
     groupId: String,
+    memberId: String = "",
+    groupInstanceId: String = null,
+    protocolType: String = "consumer",
+    protocolName: String = "consumer-range",
     metadata: Array[Byte] = Array.empty,
-    assignment: Array[Byte] = Array.empty,
-    completeRebalance: Boolean = true
-  ): (String, Int) = {
+    version: Option[Short] = None
```

Comment:
nit: How about not using an Option here and using the latest version as the default value? This would also simplify the logic later on.